polygeist/tools/cgeist/driver.cc

```diff
@@ -227,10 +227,7 @@ static void loadDialects(MLIRContext &Ctx, const bool SYCLIsDevice) {
   Ctx.getOrLoadDialect<mlir::linalg::LinalgDialect>();
   Ctx.getOrLoadDialect<mlir::polygeist::PolygeistDialect>();
   Ctx.getOrLoadDialect<mlir::vector::VectorDialect>();
-
-  if (SYCLIsDevice) {
-    Ctx.getOrLoadDialect<mlir::sycl::SYCLDialect>();
-  }
+  Ctx.getOrLoadDialect<mlir::sycl::SYCLDialect>();
```

Comment:
We are now generating SYCL operation (`sycl.addrspacecast`), even for non sycl program, so we need to always load the dialect.

Reply:
Again, we should not generate this operation in non-SYCL context IMO. This is something we should discuss.